tests/run/i11764.scala

```diff
@@ -0,0 +1,15 @@
+import deriving.Mirror
+
+sealed trait Foo
+object Foo
+
+sealed trait Bar
+
+
+@main def Test =
+  val mFoo = summon[Mirror.SumOf[Foo]]
+  assert(mFoo eq Foo)
```

Comment:
Why is the mirror the `Foo` object itself? I am confused.

Reply:
it has always been this way, the mirror of a "sum" type is its companion object

Reply:
Interesting. But that’s not visible to the users, right? See https://scastie.scala-lang.org/JE8eDCKdR4epltPeV1Z5Yw

Reply:
not unless it is loaded from tasty:
https://github.com/lampepfl/dotty/issues/13815